tt_eager/tensor/tensor_utils.cpp

```diff
@@ -402,68 +534,81 @@ Tensor get_shard_for_device(const Tensor& tensor, Device* target_device, std::op
-        if constexpr (std::is_same_v<T, MultiDeviceHostStorage>) {
-            s.insert_buffer_and_shape_for_device(buffer_index.value(), std::get<OwnedStorage>(shard.get_storage()).get_buffer(), shard.get_legacy_shape());
-        } else if constexpr (std::is_same_v<T, MultiDeviceStorage>) {
-            s.insert_buffer_and_shape_for_device(target_device, std::get<DeviceStorage>(shard.get_storage()).get_buffer(), shard.get_legacy_shape());
-        } else if constexpr (std::is_same_v<T, OwnedStorage>) {
-            s.insert_buffer(std::get<OwnedStorage>(shard.get_storage()).get_buffer());
-        } else if constexpr (std::is_same_v<T, DeviceStorage>) {
-            s.insert_buffer(std::get<DeviceStorage>(shard.get_storage()).get_buffer());
-        } else {
-            TT_FATAL(false, "Unsupported storage in insert_buffer_and_shape_for_device");
-        }
-    }, tensor_to_modify.tensor_attributes->storage);
+    std::visit(
+        [target_device, &shard, &tensor_to_modify, buffer_index](auto&& s) {
+            using T = std::decay_t<decltype(s)>;
+            if constexpr (std::is_same_v<T, MultiDeviceHostStorage>) {
+                s.insert_buffer_and_shape_for_device(
+                    buffer_index.value(),
+                    std::get<OwnedStorage>(shard.get_storage()).get_buffer(),
+                    shard.get_legacy_shape());
+            } else if constexpr (std::is_same_v<T, MultiDeviceStorage>) {
+                s.insert_buffer_and_shape_for_device(
+                    target_device, std::get<DeviceStorage>(shard.get_storage()).get_buffer(), shard.get_legacy_shape());
+            } else if constexpr (std::is_same_v<T, OwnedStorage>) {
+                s.insert_buffer(std::get<OwnedStorage>(shard.get_storage()).get_buffer());
+            } else if constexpr (std::is_same_v<T, DeviceStorage>) {
+                s.insert_buffer(std::get<DeviceStorage>(shard.get_storage()).get_buffer());
+            } else {
+                TT_FATAL(false, "Unsupported storage in insert_buffer_and_shape_for_device");
+            }
+        },
+        tensor_to_modify.tensor_attributes->storage);
 }
 
-
 Tensor copy_borrowed_tensor_in_async_mode(Device* worker, const Tensor& tensor) {
     // When using async mode, tensors with borrowed storage cannot be passed to workers.
     // They need to be copied to owned storage before being passed to the worker.
     ZoneScopedN("ConvertBorrowedToOwned");
     // Tensor has workers (on device) or runtime mode is synchronous or tensor has multiple buffers.
     // No need to check for borrowed storage.
-    if (worker->get_worker_mode() == WorkExecutorMode::SYNCHRONOUS or
-        tensor.get_workers().size() or
-        tensor.tensor_attributes->tensor_populated.size() > 1) return tensor;
+    if (worker->get_worker_mode() == WorkExecutorMode::SYNCHRONOUS or tensor.get_workers().size() or
+        tensor.tensor_attributes->tensor_populated.size() > 1)
+        return tensor;
```

Comment:
Can we always brace bracket even single line ifs?

Reply:
fixed